src/com.company/Main.java

```diff
@@ -0,0 +1,21 @@
+package com.company;
+
+public class Main {
+    public static void main(String[] args) {
+
+        A a1 = new A();
+        System.out.println();
+        A a2 = new A();
+        System.out.println();
+
+        B b1 = new B();
+        System.out.println();
+        B b2 = new B();
+        System.out.println();
+
+        C c1 = new C();
```

Comment:
@ruslanguliarenko you should create only C object to show initialization order chain